[Keymap] Adds ymdk/ymd09 andys8 layout

andys8 opened the pull request:
## Description

Custom layout for macropad. It shows the usage of macros with unicode, and other layouts (ISO-DE) with unicode, emojis, and git commands.


## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation


## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on keyboards/ymdk/ymd09/keymaps/andys8/config.h:
```diff
@@ -0,0 +1 @@
+#define UNICODE_SELECTED_MODES UC_LNX
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

And this needs `#pragma once` before the define.

andys8 replied:
Added license and pragma

drashna commented on keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:
```diff
@@ -0,0 +1,81 @@
+#include QMK_KEYBOARD_H
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

andys8 replied:
Added license

drashna commented on keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:
```diff
@@ -0,0 +1,81 @@
+        case GIT_STASH:
+            if (record->event.pressed) {
+                send_string("git stash ");
+                tap_code(DE_MINS);
```
If you include `sendstring_german.h`, you should be able to use a single function.

andys8 replied:
Thanks for this great hint. Makes everything a lot better.

andys8 commented:
@drashna I noticed that sending unicode strings is working in less places, than it initially did.
E.g. in the chrome address bar it's still working, but not anymore in the terminal.

Do you have an idea if this is related to the operating system, updates in qmk `master` branch, or due to changes in this PR?

## Chrome working

![image](https://user-images.githubusercontent.com/13085980/103933909-629c1200-5124-11eb-8806-c94dd1eb2c57.png)


## Terminal

![image](https://user-images.githubusercontent.com/13085980/103933930-6e87d400-5124-11eb-94f9-278f6f535f34.png)

## Qutebrowser

![image](https://user-images.githubusercontent.com/13085980/103933977-7fd0e080-5124-11eb-947c-a27e99dcb135.png)

fauxpark commented on keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:
```diff
@@ -0,0 +1,89 @@
+            break;
+        case GIT_STASH:
+            if (record->event.pressed) {
+                send_string("git stash"SS_TAP(X_ENTER));
```
```suggestion
                SEND_STRING("git stash\n");
```

andys8 replied:
Done. I'm actually surprised that `\n` is working. That looks easy.

fauxpark commented on keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:
```diff
@@ -0,0 +1,89 @@
+            break;
+        case GIT_STASH_POP:
+            if (record->event.pressed) {
+                send_string("git stash pop"SS_TAP(X_ENTER));
```
```suggestion
                SEND_STRING("git stash pop\n");
```

andys8 replied:
✅

fauxpark commented on keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:
```diff
@@ -0,0 +1,89 @@
+            break;
+        case GIT_COMMIT:
+            if (record->event.pressed) {
+                send_string("git add -A && git commit -a"SS_TAP(X_ENTER));
```
```suggestion
                SEND_STRING("git add -A && git commit -a\n");
```

andys8 replied:
✅

fauxpark commented on keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:
```diff
@@ -0,0 +1,89 @@
+            break;
+        case COPY_PASTE:
+            if (record->event.pressed) {
+                send_string(SS_LCTL("c"));
+            } else {
+                send_string(SS_LCTL("v"));
```
```suggestion
                tap_code16(C(KC_C));
            } else {
                tap_code16(C(KC_V));
```

andys8 replied:
This one is also an improvement. Thanks ✅

drashna commented:
Unicode or the like is a very tricky subject, because it is NOT handled consistently. Not between OS's, and sometimes, not even between programs. I don't really have a good answer for you, aside from tinkering and find what works.   

That said, there is the unicode feature in QMK which may work better.

andys8 commented:
I dropped unicode handling all together, because it wasn't reliable with different systems and applications. I'm going to handle this in the system. I added the suggested code improvements, thanks.

drashna commented:
Thanks!
